.github/workflows/main.yml

```diff
@@ -37,6 +37,24 @@ jobs:
         cd server
         poetry install
 
+    - name: Setup Node.js
```

Comment:
Is this an upstream merge from somewhere?

Reply:
This is added here in this PR (_Add client building steps to server-side testing on our CI as templates are needed_).  Basically, there are some new tests (`server/portal/apps/site_search/views_unit_test.py`) to make sure the setup_complete values are correct in the responses for `/search` for authenticated/anonymous users. These tests need the html template produced during our client install/build. So I had to add the client install/build steps to the `Server_Side_Unit_Tests` in our workflow.

I wanted to clean things up as now those same "lets install client" steps are in `Server_Side_Unit_Tests`, `Client_Side_Unit_Tests` and `Client_Side_Linting` using a `composite` action.  It didn't work 😞 .  See the last few commits in this branch and this note (https://github.com/TACC/Core-Portal/pull/358#issuecomment-789277897 )for why it probably didn't work.